src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -363,14 +363,17 @@ private void InitializeOutputFile(TOptions analyzeOptions, TContext context, ISe
 
                         SarifLogger sarifLogger;
 
-                        _run = new Run()
+                        _run = new Run();
+
+                        if (!string.IsNullOrWhiteSpace(analyzeOptions.AutomationId)
+                        || !string.IsNullOrWhiteSpace(analyzeOptions.AutomationGuid))
                         {
-                            AutomationDetails = new RunAutomationDetails
+                            _run.AutomationDetails = new RunAutomationDetails
```

Comment:
>new RunAutomationDetails [](http://example.com/codeflow?start=53&length=24)

The issue is, if there is no AutomationId and AutomationGuid, we don't need to create a instance here.
instance not null with all property null will still get written with current code:
SerializeIfNotNull(_run.AutomationDetails, "automationDetails");